### PR TITLE
Meta and metaDetail link to player and small fixes in deep links

### PR DESCRIPTION
**Pull request by core1024**

### CODE REVIEW

File: stremio-deeplinks/src/lib.rs [hunk 1]
```diff
@@ -140,16 +144,54 @@ impl From<&MetaItemPreview> for MetaItemDeepLinks {
...
+                // video id are formed like that: yt_id:YT_CHANNEL_ID:YT_VIDEO_ID
+                default_video_id.split(':').nth(2).map(|video_id| {
+                    format!(
+                        "#/player/{}/{}/{}/{}/{}/{}",
+                        utf8_percent_encode(
+                            &base64::encode(
+                                gz_encode(
+                                    serde_json::to_string(&Stream {
+                                        source: StreamSource::YouTube {
+                                            yt_id: video_id.to_string()
```
Comment (nklhtv) [inline stremio-deeplinks/src/lib.rs:166]: we used to_owned on every other place, change this for consistency

**Review by nklhtv — CHANGES_REQUESTED** [review]

File: stremio-deeplinks/src/lib.rs [hunk 2]
```diff
@@ -140,16 +144,54 @@ impl From<&MetaItemPreview> for MetaItemDeepLinks {
                         "#/metadetails/{}/{}/{}",
                         utf8_percent_encode(&item.r#type, URI_COMPONENT_ENCODE_SET),
                         utf8_percent_encode(&item.id, URI_COMPONENT_ENCODE_SET),
-                        utf8_percent_encode(&video_id, URI_COMPONENT_ENCODE_SET)
+                        utf8_percent_encode(video_id, URI_COMPONENT_ENCODE_SET)
                     )
                 }),
+            player: if item.id.starts_with(YOUTUBE_PREFIX) {
```
[the remark this answers is not in the document]
Comment (core1024) [reply stremio-deeplinks/src/lib.rs]: Done

File: stremio-deeplinks/src/lib.rs (hunk 1 above)
Comment (core1024) [reply stremio-deeplinks/src/lib.rs]: Changed

**Review by nklhtv — APPROVED** [review]

File: stremio-deeplinks/src/lib.rs [hunk 3]
```diff
@@ -140,16 +144,54 @@ impl From<&MetaItemPreview> for MetaItemDeepLinks {
...
+                Some(())
+            } else {
+                None
+            }
+            .zip(item.behavior_hints.default_video_id.clone())
```
[the remark this answers is not in the document]
Comment (core1024) [reply stremio-deeplinks/src/lib.rs]: It works with `and_then` instead of `map`

**Pull request merged**